cdap-ui/app/services/dashboard/dashboardhelper.js

```diff
@@ -0,0 +1,94 @@
+angular.module(PKG.name + '.services')
+  .factory('DashboardHelper', function (MyDataSource, MyChartHelpers, MyMetricsQueryHelper) {
+    var dataSrc = new MyDataSource();
+
+    function startPolling (widget) {
+      dataSrc.poll({
+        _cdapPath: '/metrics/query',
+        method: 'POST',
+        interval: widget.settings.interval,
+        body: MyMetricsQueryHelper.constructQuery(
+          'qid',
+          MyMetricsQueryHelper.contextToTags(widget.metric.context),
+          widget.metric
+        )
+      }, function (res) {
+
+        widget.formattedData = __formatData(res, widget);
+
+        widget.pollId = res.__pollId__;
```

Comment:
Referring to this comment (https://github.com/caskdata/cdap/pull/3079/files#r33823887)